enhancements/olm/optional-serviceaccount-field.md

```diff
@@ -0,0 +1,247 @@
+---
+title: olmv1-optional-serviceaccount-field
+authors:
+  - "@rashmigottipati"
+reviewers:
+  - "@grokspawn"
+  - "@joelanford"
+  - "@trgeiger"
+approvers:
+  - "@joelanford"
+api-approvers:
+  - "@everettraven"
+creation-date: 2025-11-25
+last-updated: 2025-11-25
+tracking-link:
+  - https://issues.redhat.com/browse/OPRUN-4144
+replaces:
+superseded-by:
+---
+
+# OLMv1: Make ServiceAccount Field Optional in the ClusterExtension API
+
+## Summary
+
+One of the core design principles of OLMv1 is Secure By Default. This enhancement aims at upholding and also improving upon that principle while addressing a usability issue.
+
+The proposal is to make the `spec.serviceAccount` field in the ClusterExtension API optional in Tech Preview. For extensions without a ServiceAccount, OLMv1 uses a synthetic identity with zero permissions and relies on Kubernetes impersonation, allowing administrators to explicitly grant the necessary privileges. For extensions with a ServiceAccount, OLMv1 continues to use token based authentication, preserving backward compatibility.
```

Comment:
> For extensions with a ServiceAccount, OLMv1 continues to use token based authentication, preserving backward compatibility.

Why can you not switch the existing to impersonation? From a permissions perspective OLM would need to be able to impersonate service accounts. Does it not have that permission today?